scipy/fft/_pocketfft/pocketfft_hdronly.h

```diff
@@ -2630,11 +2630,13 @@ template<typename T0> class T_dcst23
         if (!cosine)
           for (size_t k=0, kc=N-1; k<kc; ++k, --kc)
             std::swap(c[k], c[kc]);
-        if (ortho) c[0]*=sqrt2*T0(0.5);
+        if (ortho)
+          cosine ? c[0]*=sqrt2*T0(0.5) : c[N-1]*=sqrt2*T0(0.5);
         }
       else
         {
-        if (ortho) c[0]*=sqrt2;
+        if (ortho)
+          cosine ? c[0]*=sqrt2 : c[N-1]*=sqrt2;
```

Comment:
It would be good to fix these on the upstream repo as well, then we can sync all the other changes in the last few years.

Reply:
Sure, the PR there is already prepared! I just wanted to get positive feedback first.